File: api/src/main/java/club/yunzhi/log/entity/Semester.java
package club.yunzhi.log.entity;

import javax.persistence.*;

/**
 * 学期实体
 */
@Entity
public class Semester {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long id;

    /**
     * 名称
     */
    private String name;

    private Long startTime;

    private Long endTime;

    public Long getStartTime() {
        return startTime;
    }

    public void setStartTime(Long startTime) {
        this.startTime = startTime;
    }

    public Long getEndTime() {
        return endTime;
    }

    public void setEndTime(Long endTime) {
        this.endTime = endTime;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }


}

File: web/src/entity/semester.ts
/**
 * 学期
 */

export class Semester {
  /**
   * id
   */
  id?: number;
  /**
   * 名称
   */
  name?: string;

  /**
   * 开始时间
   */
  startTime?: number;

  /**
   * 结束时间
   */
  endTime?: number;

  constructor(data = {} as {
    id?: number,
    name?: string,
    startTime?: number,
    endTime?: number
  }) {
    this.id = data.id;
    this.name = data.name;
    this.startTime = data.startTime;
    this.endTime = data.endTime;

  }

}
